test/functional/test_framework/authproxy.py

```diff
@@ -117,23 +117,36 @@ def get_request(self, *args, **argsn):
             params = dict(args=args, **argsn)
         else:
             params = args or argsn
-        return {'version': '1.1',
+        return {'jsonrpc': '2.0',
                 'method': self._service_name,
                 'params': params,
                 'id': AuthServiceProxy.__id_count}
 
     def __call__(self, *args, **argsn):
         postdata = json.dumps(self.get_request(*args, **argsn), default=serialization_fallback, ensure_ascii=self.ensure_ascii)
         response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
-        if response['error'] is not None:
-            raise JSONRPCException(response['error'], status)
-        elif 'result' not in response:
-            raise JSONRPCException({
-                'code': -343, 'message': 'missing JSON-RPC result'}, status)
-        elif status != HTTPStatus.OK:
-            raise JSONRPCException({
-                'code': -342, 'message': 'non-200 HTTP status code but no JSON-RPC error'}, status)
+        # For backwards compatibility tests, accept JSON RPC 1.1 responses
+        if 'jsonrpc' not in response:
+            if response['error'] is not None:
+                raise JSONRPCException(response['error'], status)
+            elif 'result' not in response:
+                raise JSONRPCException({
+                    'code': -343, 'message': 'missing JSON-RPC result'}, status)
+            elif status != HTTPStatus.OK:
+                raise JSONRPCException({
+                    'code': -342, 'message': 'non-200 HTTP status code but no JSON-RPC error'}, status)
+            else:
+                return response['result']
         else:
+            assert response['jsonrpc'] == '2.0'
+            if status != HTTPStatus.OK:
```

Comment:
This would also throw if 204 was received.  This seems ok because none of our tests should send a "notification" (i.e. without `id` param), so 204 should not be received.

Reply:
There also would be no `error` or `result` fields in the 204 case! I think it's self explanatory that the authproxy call function expects a very specific type of response.